core/src/lib.rs

```diff
@@ -5,6 +5,10 @@
 #[cfg(feature = "alloc")]
 mod alloc;
 mod entry_point;
+#[cfg(all(feature = "heap", not(feature = "alloc")))]
+pub mod heap;
+#[cfg(all(feature = "heap", feature = "alloc"))]
+mod heap;
```

Comment:
Just another idea how to remove the number of switches... :blush:

```rust
#[cfg(feature = "heap")]
mod heap;
...
#[cfg(not(feature = "alloc")]
pub use heap;
```

Reply:
If neither `heap` nor `alloc` are active (fully custom allocator implementation), then we don't want this module to exist at all.

Also, when `heap = enabled && alloc = disabled` your proposal would declare the `heap` module twice.

Reply:
The second one is a `use` statement, so it's not declared twice.

Reply:
Oh I missed that.

There is still the problem of `heap = disabled && alloc = disabled` though.